base/env.jl

```diff
@@ -161,7 +161,7 @@ function parse_bool_env(name::String, val::String = ENV[name]; throw::Bool=false
     end
 end
 
-getindex(::EnvDict, k::AbstractString) = access_env(k->throw(KeyError(k)), k)
+getindex(::EnvDict, k::AbstractString) = access_env(k->throw(KeyError(ENV, k)), k)
```

Comment:
```suggestion
getindex(ENV::EnvDict, k::AbstractString) = access_env(k->throw(KeyError(ENV, k)), k)
```

Reply:
Thanks for catching this!

Reply:
The assumption of `::EnvDict` referring to the global `ENV` is also present in other functions (not written by you). I wonder if it deserve a separate issue? I guess, `EnvDict` structure is a private API, still, if someone creates its own instance it basically won't really work properly with other functions anyway.

Reply:
`EnvDict` is a singleton type, so it is fine to refer to "the global `ENV`" as it is the only instance that can exist anyway.